### DISCUSSION TITLE
How to assign Value to final userProvider = Provider<User>((ref) {   throw UnimplementedError(); });

### DISCUSSION THREAD

**Discussion by ul-rotha** [rrousselGit/riverpod#720] [Q&A]
final userProvider = Provider<User>((ref) {
  throw UnimplementedError();
});

how to assign new value to => (userProvider) after get data from web api by clicking onPress button login.

**Comment by rrousselGit** [comment]
That's a bit unclear. But chances are you don't want to use `Provider`. Instead consider `StateProvider`.
